src/main/opfs-targeted-recovery.mjs

```
@@ -103,22 +103,58 @@ async function repairDocument(instance, documentId) {
 
-async function dropWhitespaceRows(instance) {
+// Byte ranges already deleted on disk, grouped by index. A changelog "D"
+// operation is `[indexId, position, "D", [indexableString, start, end]]`; the
+// [start, end] pair identifies the row's bytes independently of its position,
+// which is what lets a peer's drop be recognised across tabs.
+function collectDroppedRanges(operationsByIndexId) {
+  const rangesByIndexId = new Map();
+  for (const [indexId, operations] of operationsByIndexId) {
+    const ranges = new Set();
+    for (const operation of operations) {
+      if (operation[2] === "D") ranges.add(`${operation[3][1]}:${operation[3][2]}`);
```

ESLint

[error] Insert `⏎·······`

(prettier/prettier)